datastore/api/snippets.py

```diff
@@ -0,0 +1,820 @@
+    # [END key_with_parent]
+
+    return key
+
+
+def key_with_multilevel_parent(client):
+    # [START key_with_multilevel_parent]
+    key = client.key(
+        'User', 'alice',
+        'TaskList', 'default',
+        'Task', 'sample_task')
+    # [END key_with_multilevel_parent]
+
+    return key
+
+
+def basic_entity(client):
+    # [START basic_entity]
+    task = datastore.Entity()
+    task.update({
+        'type': 'Personal',
+        'done': False,
+        'priority': 4,
+        'description': 'Learn Cloud Datastore'
+    })
+    # [END basic_entity]
+
+    return task
+
+
+def entity_with_parent(client):
+    # [START entity_with_parent]
+    key_with_parent = client.key(
+        'TaskList', 'default', 'Task', 'sample_task')
+
+    task = datastore.Entity(key=key_with_parent)
+
+    task.update({
+        'type': 'Personal',
+        'done': False,
+        'priority': 4,
+        'description': 'Learn Cloud Datastore'
+    })
+    # [END entity_with_parent]
+
+    return task
+
+
+def properties(client):
+    # [START properties]
+    task = datastore.Entity()
+    task.update({
+        # String
+        'type': 'Personal',
+        'description': 'Learn Cloud Datastore',
+        # Date
+        'created': datetime.datetime.utcnow(),
+        # Boolean
+        'done': False,
+        # Integer
+        'priority': 4,
+        # Float
+        'percent_complete': 10.5,
+    })
+    # [END properties]
+
+    return task
+
+
+def array_value(client):
+    # [START array_value]
+    task = datastore.Entity()
+    task.update({
+        'tags': [
+            'fun',
+            'programming'
+        ],
+        'collaborators': [
+            'alice',
+            'bob'
+        ]
+    })
+    # [END array_value]
+
+    return task
+
+
+def upsert(client):
+    # [START upsert]
+    complete_key = client.key('Task', 'sample_task')
+
+    task = datastore.Entity(key=complete_key)
+
+    task.update({
+        'type': 'Personal',
+        'done': False,
+        'priority': 4,
+        'description': 'Learn Cloud Datastore'
+    })
+
+    client.put(task)
+    # [END upsert]
+
+    return task
+
+
+def insert(client):
+    # [START upsert]
+    incomplete_key = client.key('Task')
+
+    task = datastore.Entity(key=incomplete_key)
+
+    task.update({
+        'type': 'Personal',
+        'done': False,
+        'priority': 4,
+        'description': 'Learn Cloud Datastore'
+    })
+
+    client.put(task)
+    # [END upsert]
+
+    return task
+
+
+def update(client):
+    # Create the entity we're going to update.
+    upsert(client)
+
+    # [START update]
+    key = client.key('Task', 'sample_task')
+    task = client.get(key)
+
+    task['done'] = True
+
+    client.put(task)
+    # [END update]
+
+    return task
+
+
+def lookup(client):
+    # Create the entity that we're going to look up.
+    upsert(client)
+
+    # [START lookup]
+    key = client.key('Task', 'sample_task')
+    task = client.get(key)
+    # <Entity
```

Comment:
I don't think it's useful here. The node ones showed it because the properties are hidden under data.